docs/source/conf.py

```diff
@@ -425,27 +430,70 @@ def check_output(self, want, got, optionflags):
                 return False
         return True
 
+import sphinx.ext.doctest
+
+class PlatformDocTestRunner(sphinx.ext.doctest.SphinxDocTestRunner):
+    def run(self, test, compileflags=None, out=None, clear_globs=True):
+        original_optionflags = self.optionflags | test.globs.get('doctest_additional_flags', 0)
+        def filter_platform(example):
+            optionflags = original_optionflags
+            if example.options:
+                for (optionflag, val) in example.options.items():
+                    if val:
+                        optionflags |= optionflag
+                    else:
+                        optionflags &= ~optionflag
+
+            if (optionflags & WINDOWS) == WINDOWS and sys.platform != 'win32':
+                return False
+            if (optionflags & LINUX) == LINUX and sys.platform != 'linux':
```

Comment:
I wonder how this excludes FreeBSD for example. Some tests are really Linux-specific (procfs), but some others only depend on ELF or POSIX being the way things are done.

Reply:
Maybe add more platform options for those? `POSIX` sounds low effort enough to decide. We don't test on bsd or mac so this would only be for possible future testing?

Reply:
Yeah, just for the contributors I suppose.

Reply:
I've switched most of the markers to `POSIX` now instead of strictly `LINUX`.